packages/zoom-in-ui/src/components/app.js

```diff
@@ -79,12 +109,18 @@ const App = props => {
   const [theme, setTheme] = useState(
     imageCaption ? themes.twreporterTheme : themes.defaultTheme
   )
+  const [isValidImage, setImageValid] = useState()
+
+  const isImageValid = async imageLink => {
+    setImageValid(await checkIfImageIsValid(imageLink))
```

Comment:
Could we remove Promise here?
Such as (just prototyping),
```
const [isValidImage, setImageValid] = useState()
const checkIfImageIsValid = imageSrc => {
  if (typeof window === 'undefined')
      reject(new Error('window is not defined'))
    const img = new window.Image()
    img.onload = () => setImageValid(true)
    img.onerror = () => setImageValid(false)
    img.src = imageSrc
}
```

Would it be a little bit more compact?

Reply:
Updated.